stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImpl.kt

```diff
@@ -152,6 +155,33 @@ internal class QueryChannelsControllerImpl(
                 refreshChannel(event.cid)
             }
         }
+
+        if (event is UserPresenceChangedEvent) {
+            refreshMembersStateForUser(event.user)
+        }
+    }
+
+    private suspend fun refreshMembersStateForUser(newUser: User) {
+        val userId = newUser.id
+
+        val affectedChannels = _channels.value
+            .values
+            .filter { channel -> channel.users().map { user -> user.id }.contains(userId) }
+            .map { channel ->
+                val copiedChannel = channel.copy()
+                val copiedMembers = copiedChannel.members.map { member -> member.copy() }
+
+                copiedChannel.members = copiedMembers
+                copiedChannel
+            }
+
+        affectedChannels.forEach { channel ->
+            channel.members
+                .find { member -> member.user.id == userId }
+                ?.let { member -> member.user = newUser }
+        }
```

Comment:
We don't need to go through all channels multiple times to update members, it can be done on the same `map`
```suggestion
            .map { channel ->
                channel.copy(
                    members = channel.members.map { member ->
                        member.copy(user = member.user.takeUnless { it.id == userId } ?: newUser)
                    }
                )
            }
```